core/src/test/java/org/apache/iceberg/CachingCatalogTestHelper.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.github.benmanes.caffeine.cache.Ticker;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.junit.Assert;
+
+/**
+ * Utility class for accessing package-private VisibleForTesting annotated methods
+ * outside the org.apache.iceberg package.
+ */
+public class CachingCatalogTestHelper {
+
+  // TODO - Consider making this a delegator?
+  private CachingCatalogTestHelper() {
```

Comment:
Is this needed?

Reply:
I personally didn't care for this either.

The reason to have this is that the style checker complains that the `@VisibleForTesting` annotation is used for things that aren't package-private. The methods can't be package private otherwise as `TestCachingCatalog` is in org.apache.iceberg.hadoop but `CachingCatalog` is in org.apache.iceberg.

It was _much_ cleaner without this. I can try to move the tests instead?

Reply:
Another option is to make instance variables package private and then have a subclass for testing that exposes the methods publicly. Then you have a whole class that is visible for testing. That would also remove test code from the production class.

Reply:
I like that idea. I'm working on another PR today, so I'm going to leave as is for the moment and come back to this later but your idea is much cleaner. 👍

Reply:
I would be in favor of removing `CachingCatalogTestHelper` and inline the assertion checks within the tests

Reply:
Yeah that's how I had it previously, but the `@VisibleForTesting` annotations failed.

I'll definitely remove this class and then make a subclass like Ryan mentioned. This class came in sort of last minute to get around the compilation failures from using `@VisibleForTesting` and admittedly I do not like this class at all.

Reply:
This class has been removed!